library/src/main/java/com/bumptech/glide/Glide.java

```diff
@@ -137,6 +139,17 @@ public static File getPhotoCacheDir(Context context, String cacheName) {
   }
 
   /**
+   * Enable or disable the parsing of AndroidManifest.xml
+   * looking for {@link GlideModule} implementations.
+   * Must be called before accessing the Glide singleton; otherwise, has no effect.
+   */
+  public static void setModulesEnabled(boolean enabled) {
+    synchronized (Glide.class) {
```

Comment:
I'd probably add a check here to make sure that the Glide singleton hasn't already been instantiated.